includes/class-analytics.php

```diff
@@ -107,6 +107,42 @@ public static function get_events() {
 					'verticalBoundaries' => [ 100 ],
 				],
 			],
+
+			// Newsletters.
+			[
+				'id'             => 'newsletterSignup',
+				'amp_on'         => 'amp-form-submit-success',
+				'on'             => 'submit',
+				'amp_element'    => '#mailchimp_form',
+				'element'        => '.wp-block-jetpack-mailchimp form',
+				'event_name'     => 'newsletter signup',
+				'event_label'    => 'success',
+				'event_category' => 'NTG newsletter',
+			],
+			[
+				'id'             => 'newsletterImpression',
+				'on'             => 'visible',
+				'amp_element'    => '.newspack-newsletter-prompt-overlay',
+				'element'        => '.wp-block-jetpack-mailchimp form',
+				'event_name'     => 'newsletter modal impression 1', // 1: overlay/lightbox
+				'event_label'    => get_the_title(),
+				'event_category' => 'NTG newsletter',
+				'visibilitySpec' => [
+					'totalTimeMin' => 500,
+				],
+			],
+			[
+				'id'             => 'newsletterImpression',
+				'on'             => 'visible',
+				'amp_element'    => '.newspack-newsletter-prompt-inline',
+				'element'        => '.wp-block-jetpack-mailchimp form',
+				'event_name'     => 'newsletter modal impression 2', // 2: inline prompt
+				'event_label'    => get_the_title(),
+				'event_category' => 'NTG newsletter',
+				'visibilitySpec' => [
+					'totalTimeMin' => 500,
+				],
```

Comment:
The timeout will not be handled in non-AMP – perhaps we should drop it, so that AMP and non-AMP reporting is as similar as possible?

Reply:
I removed it for now.

Reply:
As mentioned in another comment, this seems to be needed to avoid false impressions on hidden campaigns. I think the better solution would be to recreate the timeout logic for non-AMP events as well.